src/masternodes/icxorder.cpp

```diff
@@ -0,0 +1,248 @@
+#include <masternodes/icxorder.h>
+
+/// @attention make sure that it does not overlap with other views !!!
+const unsigned char CICXOrderView::ICXOrderCreationTx           ::prefix = '1';
+const unsigned char CICXOrderView::ICXOrderKey                  ::prefix = '2';
+const unsigned char CICXOrderView::ICXOrderStatus               ::prefix = 'B';
+const unsigned char CICXOrderView::ICXMakeOfferCreationTx       ::prefix = '3';
+const unsigned char CICXOrderView::ICXMakeOfferKey              ::prefix = '4';
+const unsigned char CICXOrderView::ICXSubmitDFCHTLCCreationTx   ::prefix = '5';
+const unsigned char CICXOrderView::ICXSubmitDFCHTLCKey          ::prefix = '6';
+const unsigned char CICXOrderView::ICXSubmitDFCHTLCStatus       ::prefix = 'C';
+const unsigned char CICXOrderView::ICXSubmitEXTHTLCCreationTx   ::prefix = '7';
+const unsigned char CICXOrderView::ICXSubmitEXTHTLCKey          ::prefix = '8';
+const unsigned char CICXOrderView::ICXClaimDFCHTLCCreationTx    ::prefix = '9';
+const unsigned char CICXOrderView::ICXClaimDFCHTLCKey           ::prefix = '0';
+const unsigned char CICXOrderView::ICXCloseOrderCreationTx      ::prefix = 'A';
+
+const int CICXOrder::DEFAULT_EXPIRY = 2880;
+const int CICXOrder::TYPE_INTERNAL = 1;
+const int CICXOrder::TYPE_EXTERNAL = 2;
+const int CICXOrder::STATUS_OPEN = 0;
+const int CICXOrder::STATUS_CLOSED = 1;
+const int CICXOrder::STATUS_EXPIRED = 2;
+
+const int CICXMakeOffer::STATUS_OPEN = 0;
+
+const int CICXSubmitDFCHTLC::DEFAULT_TIMEOUT = 100;
+const int CICXSubmitDFCHTLC::STATUS_OPEN = 0;
+const int CICXSubmitDFCHTLC::STATUS_CLAIMED = 1;
+const int CICXSubmitDFCHTLC::STATUS_REFUNDED = 2;
+
+const int CICXSubmitEXTHTLC::STATUS_OPEN = 0;
+const int CICXSubmitEXTHTLC::STATUS_EXPIRED = 1;
```

Comment:
Can you use `uint8_t`

Reply:
I think I had some warning in comparison with some unsigned ints, but will use uint8_t as it is one byte!